src/utils.js

```diff
@@ -343,3 +348,42 @@ const sleep = (millis) => {
 export const publicUtils = {
     sleep,
 };
+
+/**
+ * Returns a promise that resolves to an array of urls parsed from the resource
+ * available at the provided url.
+ *
+ * @param {String} url
+ * @param {String} encoding
+ * @param {Number} offset
+ * @param {Number} limit
+ * @param {Boolean} skipDuplicates
+ * @param {RegExp} urlRegExp
+ * @returns {Promise<Array>}
+ */
+export const downloadListOfUrls = ({ url, encoding, offset, limit, skipDuplicates, urlRegExp }) => {
+    if (!url) return Promise.reject(new Error('Resource URL must be provided.'));
+    encoding = encoding || 'utf8';
+    offset = offset || 0;
+    return requestPromise.get({ url, encoding })
+        .then((string) => {
+            const urlsArr = extractUrls({ string, urlRegExp }); // not using keep commas for now
+            const limited = urlsArr.slice(offset, limit || undefined);
```

Comment:
I would remove `ofset` and `limit` parameters. User can simply call:

(await downloadListOfUrls(url)).slice(offset, limit)

so there is not a lot of value. But if we decide to keep it then I would for deduplicate and then sliced resulting array.

Reply:
Actually I'm thinking the skipping of duplicates is already implemented in RequestList, where it's done correctly using the uniqueKey (which e.g. ignores case of letters in hostname). We should duplicate this logic, so I'd remove the `skipDuplicates` param altogether here.